server/tserver/src/main/java/org/apache/accumulo/tserver/scan/ScanTask.java

```diff
@@ -33,7 +33,7 @@
 
   protected final TabletHostingServer server;
   protected AtomicBoolean interruptFlag;
-  protected ArrayBlockingQueue<Object> resultQueue;
+  protected CompletableFuture<Object> resultFuture;
```

Comment:
Is this required to be accessed by sub-classes?  Looking at the usages, it seems internal to this class - but not sure we can change now (if it is changing the API?)

Reply:
Also, do we need to mark this volatile? I don't _think_ we do but just wanted to mention it to double check with others who are more familiar with the scan code. I took a quick look and it looks like instances of ScanTask (LookupTask or NextBatchTask) should only be accessed by a TServer in a single thread since the TServer will create new instances when getting each RPC call over thrift but I could be missing something.

Reply:
I'm not sure that Future's are meant to be re-used. Maybe CompletableFuture is, but there is a lot of complexity (ref: [source](https://github.com/openjdk/jdk/blob/jdk-11+28/src/java.base/share/classes/java/util/concurrent/CompletableFuture.java#L2071) built into it because it also implements CompletionStage. That complexity is going to cost time, when in reality you just want to set a reference to the result object. I wonder if AtomicReference might be more suitable here.

Reply:
I think the main reason to use a Future or current ArrayBlockQueue is so you can wait for the result and have a timeout which is not possible with AtomicReference so I don't know that AtomicReference would work. 

https://github.com/apache/accumulo/blob/540179d1f52dcc478eee3a3ee3c5fac106736c8b/server/tserver/src/main/java/org/apache/accumulo/tserver/scan/ScanTask.java#L124

Reply:
Another option is to leave as-is, it works.

Reply:
I'd probably just leave it as is, I don't see a pressing need to actually change this.

Reply:
Okay I'll just close this for now along with the ticket. Seems like it might not be worth changing this right now.